admin/load.php

```diff
@@ -126,29 +126,38 @@ function perflab_render_modules_page() {
  * @param array  $module_settings Associative array of the module's current settings.
  */
 function perflab_render_modules_page_field( $module_slug, $module_data, $module_settings ) {
-	$base_id   = sprintf( 'module_%s', $module_slug );
-	$base_name = sprintf( '%1$s[%2$s]', PERFLAB_MODULES_SETTING, $module_slug );
-	$enabled   = isset( $module_settings['enabled'] ) && $module_settings['enabled'];
+	$base_id         = sprintf( 'module_%s', $module_slug );
+	$base_name       = sprintf( '%1$s[%2$s]', PERFLAB_MODULES_SETTING, $module_slug );
+	$enabled         = isset( $module_settings['enabled'] ) && $module_settings['enabled'];
+	$can_load_module = perflab_can_load_module( $module_slug );
 	?>
 	<fieldset>
 		<legend class="screen-reader-text">
 			<?php echo esc_html( $module_data['name'] ); ?>
 		</legend>
 		<label for="<?php echo esc_attr( "{$base_id}_enabled" ); ?>">
-			<input type="checkbox" id="<?php echo esc_attr( "{$base_id}_enabled" ); ?>" name="<?php echo esc_attr( "{$base_name}[enabled]" ); ?>" aria-describedby="<?php echo esc_attr( "{$base_id}_description" ); ?>" value="1"<?php checked( $enabled ); ?>>
+			<input type="checkbox" id="<?php echo esc_attr( "{$base_id}_enabled" ); ?>" name="<?php echo esc_attr( "{$base_name}[enabled]" ); ?>" aria-describedby="<?php echo esc_attr( "{$base_id}_description" ); ?>" value="1"<?php checked( $enabled ); ?><?php echo ( ! $can_load_module ? ' readonly onclick="return false;"' : '' ); ?>>
```

Comment:
I don't think we should have the `onclick` attribute in here - what's the purpose of that? `readonly` should be sufficient?

Reply:
I forget to add details related to this change in the original comment, Thanks for pointing it out.

If we use `readonly` than we don't block checkbox functinality because WordPress has only below CSS for the `readable` attribute and there is no other JS for it. In most cases for `readable`, they have used additional CSS property called `pointer-events: none;` but we don't have it so as per [this doc](https://stackoverflow.com/questions/12267242/how-can-i-make-a-checkbox-readonly-not-disabled) I use the return function and use `readonly` so it feels like it is disabled.

    input.readonly, input[readonly], textarea.readonly, textarea[readonly] {
        background-color: #f0f0f1;
    }

Reply:
> Why not to just disable such checkboxes using the `disabled` attribute? What is the point of having a checkbox to be readonly? This `onclick="return false;"` approach looks like a hack.

Thanks, Eugene for the concern.  If we use `disabled` then as Felix said it will not send the option when we submit the form and it also changes the checked state of the option to unchecked. Yes, It's a hack to stop the checkbox click and it will not change the check box state on submission.

@felixarntz if we go with a two checkbox approach can we go with these code changes? In the second checkbox, I removed the ID so we don't get a duplicate ID on the page.

    <input type="checkbox" id="<?php echo esc_attr( "{$base_id}_enabled" ); ?>" name="<?php echo esc_attr( "{$base_name}[enabled]" ); ?>" aria-describedby="<?php echo esc_attr( "{$base_id}_description" ); ?>" value="1"<?php checked( $enabled ); ?><?php echo ( ! $can_load_module ? ' disabled' : '' ); ?>>
    <?php
    if ( ! $can_load_module ) {
        ?>
        <input type="checkbox" name="<?php echo esc_attr( "{$base_name}[enabled]" ); ?>" aria-describedby="<?php echo esc_attr( "{$base_id}_description" ); ?>" value="1"<?php checked( $enabled ); ?>  readonly >
        <?php
    }

Reply:
Thanks. Changes applied on PR.